src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CallVariantsFromAlignedContigsSpark.java

```diff
@@ -0,0 +1,440 @@
+        final VCFHeader header = new VCFHeader();
+        header.setSequenceDictionary(referenceSequenceDictionary);
+        header.addMetaDataLine(VCFStandardHeaderLines.getInfoLine(VCFConstants.END_KEY));
+        GATKSVVCFHeaderLines.vcfHeaderLines.values().forEach(header::addMetaDataLine);
+        return header;
+    }
+
+    private static void writeVariants(final String outputPath, final List<VariantContext> variantsArrayList, final PipelineOptions pipelineOptions, final String fileName, final VCFHeader header, final SAMSequenceDictionary referenceSequenceDictionary) {
+        try (final OutputStream outputStream = new BufferedOutputStream(
+                BucketUtils.createFile(outputPath + "/" + fileName, pipelineOptions))) {
+
+            final VariantContextWriter vcfWriter = getVariantContextWriter(outputStream, referenceSequenceDictionary);
+
+            vcfWriter.writeHeader(header);
+            variantsArrayList.forEach(vcfWriter::add);
+            vcfWriter.close();
+
+        } catch (IOException e) {
+            throw new GATKException("Could not create output file", e);
+        }
+    }
+
+    private static VariantContextWriter getVariantContextWriter(final OutputStream outputStream, final SAMSequenceDictionary referenceSequenceDictionary) {
+        VariantContextWriterBuilder vcWriterBuilder = new VariantContextWriterBuilder()
+                                                            .clearOptions()
+                                                            .setOutputStream(outputStream);
+
+        if (null != referenceSequenceDictionary) {
+            vcWriterBuilder = vcWriterBuilder.setReferenceDictionary(referenceSequenceDictionary);
+        }
+        // todo: remove this when things are solid?
+        vcWriterBuilder = vcWriterBuilder.setOption(Options.ALLOW_MISSING_FIELDS_IN_HEADER);
+        for (Options opt : new Options[]{}) {
+            vcWriterBuilder = vcWriterBuilder.setOption(opt);
+        }
+
+        return vcWriterBuilder.build();
+    }
+
+    private static Iterable<BreakpointAlignment> assembledBreakpointsFromAlignmentRegions(final byte[] contigSequence, final Iterable<AlignmentRegion> alignmentRegionsIterable, final Integer minAlignLength) {
+        final List<AlignmentRegion> alignmentRegions = IterableUtils.toList(alignmentRegionsIterable);
+        if (alignmentRegions.size() > 1) {
+            alignmentRegions.sort(Comparator.comparing(a -> a.startInAssembledContig));
+        }
+        return getBreakpointAlignmentsFromAlignmentRegions(contigSequence, alignmentRegions, minAlignLength);
+    }
+
+    @VisibleForTesting
+    private static VariantContext filterBreakpointsAndProduceVariants(final Tuple2<BreakpointAllele, Iterable<Tuple2<Tuple2<String, String>, BreakpointAlignment>>> assembledBreakpointsPerAllele, final Broadcast<ReferenceMultiSource> broadcastReference) throws IOException {
```

Comment:
Could this method's name be changed? It doesn't appear to be filtering any breakpoints right now (although I believe in the future it could), or breakpoints are supposed to be filtered if they have zero `highMqMappings`?